sphinx/ext/autodoc/__init__.py

```diff
@@ -693,17 +693,26 @@ def document_members(self, all_members: bool = False) -> None:
         member_order = self.options.member_order or \
             self.env.config.autodoc_member_order
         if member_order == 'groupwise':
-            # sort by group; relies on stable sort to keep items in the
-            # same group sorted alphabetically
-            memberdocumenters.sort(key=lambda e: e[0].member_order)
-        elif member_order == 'bysource' and self.analyzer:
-            # sort by source order, by virtue of the module analyzer
-            tagorder = self.analyzer.tagorder
-
-            def keyfunc(entry: Tuple[Documenter, bool]) -> int:
-                fullname = entry[0].name.split('::')[1]
-                return tagorder.get(fullname, len(tagorder))
-            memberdocumenters.sort(key=keyfunc)
+            # sort by group; alphabetically within groups
+            memberdocumenters.sort(key=lambda e: (e[0].member_order, e[0].name))
+        elif member_order == 'bysource':
+            if self.analyzer:
+                # sort by source order, by virtue of the module analyzer
+                tagorder = self.analyzer.tagorder
+
+                def keyfunc(entry: Tuple[Documenter, bool]) -> int:
+                    fullname = entry[0].name.split('::')[1]
+                    return tagorder.get(fullname, len(tagorder))
+                memberdocumenters.sort(key=keyfunc)
+            else:
+                # Assume that member discovery order matches source order.
+                # This is a reasonable assumption in Python 3.6 and up, where
+                # module.__dict__ is insertion-ordered.
+                pass
+        elif member_order == 'alphabetic':
```

Comment:
I noticed our doc says the correct value is `'alphabetical'`. And I feel it is better wording.
https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_member_order

Reply:
That may be, but all the tests use this spelling

Reply:
I posted https://github.com/sphinx-doc/sphinx/pull/7679 to migrate the old name to new one with a warning. I suppose it helps to migrate to new options.

Reply:
```suggestion
        elif member_order == 'alphabetical':
```